### PR TITLE
BF: Fixed `--target` and `--user` argument conflict when installing plugins

### PR DESCRIPTION
**Pull request by mdcutone**
Fixes an issue where the argument `--target` and `--user` were both being specified in when calling `pip`, causing an error.

### CODE REVIEW

File: psychopy/tools/pkgtools.py [hunk 1]
```diff
@@ -240,27 +240,24 @@ def installPackage(
...
+
 
     # construct the pip command and execute as a subprocess
     cmd = [sys.executable, "-m", "pip", "install", package]
 
     # optional args
     if target is None:  # default to user packages dir
-        cmd.append('--prefix')
-        cmd.append(prefs.paths['packages'])
+        cmd.append('--user')
```
Comment (mh105) [inline psychopy/tools/pkgtools.py:254]: I'm not sure this is correct. Doesn't this bypass the whole if statement checking if in virtual environment? This will also result in duplicated `--user` flags if not within virtual env.

File: psychopy/tools/pkgtools.py [hunk 2]
```diff
@@ -240,27 +240,24 @@ def installPackage(
             and error from the subprocess.
         If `awaited=False`:
             Returns the job (thread) which is running the install.
-    """
-    if target is None:
-        target = prefs.paths['userPackages']
```
Comment (mh105) [inline psychopy/tools/pkgtools.py:245]: I think it makes sense to first check `prefs.paths['userPackages']` trying to set `target`. Should this actually get removed?

File: psychopy/tools/pkgtools.py [hunk 3]
```diff
@@ -240,27 +240,24 @@ def installPackage(
...
-            'Cannot install package "{}" to "{}", directory does not '
-            'exist.'.format(package, target))
+
 
     # construct the pip command and execute as a subprocess
     cmd = [sys.executable, "-m", "pip", "install", package]
 
     # optional args
     if target is None:  # default to user packages dir
-        cmd.append('--prefix')
```
Comment (mh105) [inline psychopy/tools/pkgtools.py:261]: This then tries to use default user packages dir as `--prefix`, which is a different behavior from using `--user`. 
Should this actually get removed?

I think the correct order of checking should use another if. It's easier for me to make another PR to show what I am thinking.

Comment (mdcutone) [reply psychopy/tools/pkgtools.py:261]: We configure the package directory as the user site since we can just pass that down as an environment variable to the subprocess.

Comment (mh105) [reply psychopy/tools/pkgtools.py:261]: Are you referring to the [`[PYTHONUSERBASE]`](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONUSERBASE) environment variable? It looks like the pip documentation recommends using `--user` over `--prefix`. However, the `--user` scheme has an annoying directory structure on macOS framework python that does not distinguish among Python versions. I have opened #6769 for this.
